Source/Parallelization/GuardCellManager.cpp

```diff
@@ -0,0 +1,160 @@
+#include "GuardCellManager.H"
+#include "NCIGodfreyFilter.H"
+#include <AMReX_Print.H>
+
+using namespace amrex;
+
+int
```

Comment:
Could you add a Doxygen string mentioning that this calculates the number of guard cells needed for different parts of the PIC loop?

Reply:
Also, this is a minor point, but I think it would be a bit cleaner if the constructor does not return an `int`, but is `void` instead.
It seems to me that the returned value (`nJ_buffer`) could be easily obtained in another way (`guard.ng_alloc_J.max()`) after the `guardCellManager` is constructed.

Reply:
I added an extensive Doxygen documentation, and made this function `void`.